src/Uno.UI/UI/Xaml/FrameworkTemplatePool.cs

```diff
@@ -346,7 +351,9 @@ internal static void PropagateOnTemplateReused(object instance)
 			// If DataContext is not null, it means it has been explicitly set (not inherited). Resetting the view could push an invalid value through 2-way binding in this case.
 			if (instance is IFrameworkTemplatePoolAware templateAwareElement && (instance as IFrameworkElement)!.DataContext == null)
 			{
+				IsRecycling = true;
```

Comment:
Why is it a static property?

I would imagine this to be a property in `IFrameworkTemplatePoolAware` and set it on the specific instance whose template is being recycled

Reply:
The reason is that it is the `BindingExpression` that is then checking for whether a recycling is in progress, which makes it a bit tricky to get the owner view to check if it being recycled. Currently, the only thing the template aware types do is they change a single property, and it all needs to happen on the UI thread, so this seems quite safe. If I did it instance based, all two way bindings would then first need to get the owner somehow (by accessing a weakref probably) and then checking if it is in fact instace of `IFrameworkTemplatePoolAware` and then check the property. So this simpler approach seems to me better perf-wise